src/templates/plugin.main.js

```diff
@@ -351,31 +351,37 @@ export default async (context) => {
   // Initialize locale and fallbackLocale as vue-i18n defaults those to 'en-US' if falsey
   app.i18n.locale = ''
   app.i18n.fallbackLocale = vueI18nOptions.fallbackLocale || ''
-  extendVueI18nInstance(app.i18n)
-  const resolveBaseUrlOptions = {
-    differentDomains: options.differentDomains,
-    normalizedLocales: options.normalizedLocales
-  }
-  app.i18n.__baseUrl = resolveBaseUrl(options.baseUrl, context, '', resolveBaseUrlOptions)
-  app.i18n.__onNavigate = onNavigate
-
-  Vue.prototype.$nuxtI18nSeo = nuxtI18nSeo
-  Vue.prototype.$nuxtI18nHead = nuxtI18nHead
 
   if (store) {
     // Inject in store.
     store.$i18n = app.i18n
 
     if (store.state.localeDomains) {
-      for (const locale of app.i18n.locales) {
+      let index = 0
+      for (const locale of options.normalizedLocales) {
         if (typeof (locale) === 'string') {
           continue
         }
-        locale.domain = store.state.localeDomains[locale.code]
+        const domain = store.state.localeDomains[locale.code]
+        locale.domain = domain
```

Comment:
The options defined in the options import are not really supposed to be mutated since that affects the whole app and not only the current request.

Maybe it makes sense in this case bu will have to look closer into the logic when I have more time. But possibly we should be storing the updated info somewhere on the instance or in the store instead of mutating the original options.

Reply:
I agree that mutating the options is not great. But in the default flow where you only have to deal with static domains you can define them in the configuration file. You end up having the domain property available in the options directly. Since the options import is the one mainly used for all the following tasks and tests I ended up mutating it to keep the logic the same as defining the domains directly in the configuration file. Of course I'm happy to find a better way to do it 😄 This PR was mainly to raise the issue with some description of the problem.

Reply:
Sorry for me forgetting about this PR.

I guess it's acceptable after all since the domains are not really meant to change per-request but rather be defined per-app so it should be fine if the options are mutated.